internal/expr/parser.go

```diff
@@ -109,6 +109,62 @@ func (p *Parser) add(part queryPart) {
 	p.partStart = p.pos
 }
 
+// removeComments passes over the input removing comments.
+// It does not allow for nested comments.
+func (p *Parser) removeComments() error {
+	var res bytes.Buffer
+	var textStart = p.pos
+	var textEnd = p.pos
+
+	for p.pos < len(p.input) {
+		textEnd = p.pos
+		// Ignore comment chars in string literals.
+		if ok, err := p.skipStringLiteral(); err != nil {
+			return err
+		} else if ok {
+			continue
+		}
+
+		if p.skipByte('-') || p.skipByte('/') {
+			startPos := p.pos - 1
```

Comment:
`startPos` doesn't seem to be used anywhere else besides the line below. Might instead do `c := p.input[p.pos-1]`

Reply:
True, changed.